File: apps/pos/native-windows/DesktopKitchenPOS/Utilities/DateFormatters.cs
using System.Globalization;

namespace DesktopKitchenPOS.Utilities;

public static class DateFormatters
{
    public static DateTime? ParseISO(string? dateString)
    {
        if (string.IsNullOrEmpty(dateString)) return null;
        if (DateTime.TryParse(dateString, CultureInfo.InvariantCulture,
            DateTimeStyles.RoundtripKind, out var dt))
            return dt.ToUniversalTime();
        return null;
    }

    public static int ElapsedSeconds(string? dateString)
    {
        var dt = ParseISO(dateString);
        if (dt == null) return 0;
        return (int)(DateTime.UtcNow - dt.Value).TotalSeconds;
    }

    public static string FormatElapsed(int seconds)
    {
        var mins = seconds / 60;
        var secs = seconds % 60;
        return mins > 0 ? $"{mins}m {secs}s" : $"{secs}s";
    }

    public static string FormatTime(string? dateString)
    {
        var dt = ParseISO(dateString);
        if (dt == null) return "";
        var local = dt.Value.ToLocalTime();
        var hour = local.Hour % 12 == 0 ? 12 : local.Hour % 12;
        var ampm = local.Hour < 12 ? "AM" : "PM";
        return $"{hour}:{local.Minute:D2} {ampm}";
    }
}

File: apps/pos/native-android/app/src/main/java/com/desktopkitchen/pos/utilities/DateFormatters.kt
package com.desktopkitchen.pos.utilities

import java.time.Instant
import java.time.ZoneId
import java.time.format.DateTimeFormatter
import java.time.format.DateTimeParseException

object DateFormatters {
    private val isoParser = DateTimeFormatter.ISO_DATE_TIME

    fun parseISO(dateString: String): Instant? {
        return try {
            Instant.from(isoParser.parse(dateString))
        } catch (e: DateTimeParseException) {
            try {
                Instant.parse(dateString)
            } catch (e2: Exception) {
                null
            }
        }
    }

    fun elapsedSeconds(from: String): Int {
        val instant = parseISO(from) ?: return 0
        return ((System.currentTimeMillis() - instant.toEpochMilli()) / 1000).toInt()
    }

    fun formatElapsed(seconds: Int): String {
        val mins = seconds / 60
        val secs = seconds % 60
        return if (mins > 0) "${mins}m ${secs}s" else "${secs}s"
    }

    fun formatTime(dateString: String): String {
        val instant = parseISO(dateString) ?: return ""
        val localTime = instant.atZone(ZoneId.systemDefault()).toLocalTime()
        val hour = if (localTime.hour % 12 == 0) 12 else localTime.hour % 12
        val ampm = if (localTime.hour < 12) "AM" else "PM"
        return String.format("%d:%02d %s", hour, localTime.minute, ampm)
    }
}
